### PR TITLE
Fix: mock.lua callback arg passing, time.lua time argument

### PR DESCRIPTION
**Pull request by insalitygames**
mock.lua: fix args passing if first args was nil
time.lua: os.time has the additional args

### CODE REVIEW

File: deftest/mock/mock.lua [hunk 1]
```diff
@@ -92,10 +92,8 @@ function mock.mock(module)
 					mock_fn.calls = mock_fn.calls + 1
 					local arg = {...}
 
-					if #arg > 0 then
-						for i=1,#arg do
-							mock_fn.params[i] = arg[i]
-						end
+					for index, value in pairs(arg) do
```
Comment (britzl) [inline deftest/mock/mock.lua:96]: Hmm, perhaps we need to make sure that mock_fn.params is cleared before use (or replaced with a new empty table)? This was a potential problem with the previous implementation as well.

File: deftest/mock/time.lua [hunk 2]
```diff
@@ -9,9 +9,9 @@ function M.mock()
 	mock.mock(socket)
 
 	mock_time = socket.gettime.original()
-	os.time.replace(function()
-		if not mock_time then
-			return os.time.original()
+	os.time.replace(function(time_table)
```
Comment (britzl) [inline deftest/mock/time.lua:12]: Nice improvement!

**Review by britzl — CHANGES_REQUESTED** [review]
See question about replacing params

File: deftest/mock/mock.lua (hunk 1 above)
Comment (Insality) [reply deftest/mock/mock.lua:96]: Time to fix it!
Agree, the new empty table here is enough :)

**Review by Insality — APPROVED** [review]

**Review by britzl — APPROVED** [review]

**Pull request merged**